src/components/Slider/Slider.js

```diff
@@ -0,0 +1,102 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+// Style
+import Container from './Container';
+import Scale from './Scale';
+import Label from './Label';
+import Input from './Input';
+
+/**
+ * @component
+ */
+class Slider extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      index: 0
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.onChangeEnd = this.onChangeEnd.bind(this);
+    this.onScaleClick = this.onScaleClick.bind(this);
+  }
+
+  componentWillMount() {
+    const { options, value } = this.props;
+
+    if (!options.length) {
+      throw new Error('options is empty');
+    }
+
+    const index = options.findIndex(o => o.value === value);
+    this.setState({ index });
+  }
+
+  onChange(event) {
+    this.setState({ index: event.target.value });
+    const index = Math.round(event.target.value);
+    const { value } = this.props.options[index];
+
+    if (value !== this.props.value) {
+      this.props.onChange(value);
+    }
+  }
+
+  onChangeEnd(event) {
+    const index = Math.round(event.target.value);
+    this.setState({ index });
+  }
+
+  onScaleClick(index) {
+    this.setState({ index });
+    const { value } = this.props.options[index];
+    this.props.onChange(value);
+  }
+
+  render() {
+    const { options } = this.props;
+
+    return (
+      <Container>
+        <Input
+          type="range"
+          value={this.state.index}
+          onChange={this.onChange}
+          onMouseUp={this.onChangeEnd}
+          max={options.length - 1}
+          step={0.01}
+        />
+        <Scale length={options.length}>
+          {
+            options.map((option, index) => (
```

Comment:
move this to class method. Something like `_renderOption(option, index)`, so you can use it as `options.map(this._ _renderOption)`